Libraries/LibWeb/StorageAPI/StorageBottle.cpp

```diff
@@ -85,4 +86,68 @@ RefPtr<StorageBottle> obtain_a_local_storage_bottle_map(HTML::EnvironmentSetting
     return obtain_a_storage_bottle_map(StorageType::Local, environment, identifier);
 }
 
+size_t StorageBottle::size() const
+{
+    if (m_storage_type == StorageType::Local)
+        return m_page->client().page_did_request_storage_keys("localStorage"_string, m_storage_key.to_string()).size();
```

Comment:
localStorage should not be hardcoded here - we should be storing what is passed through `obtain_a_local_storage_bottle_map` and reusing it here

Reply:
I can't comment on the line, but I think this is might be breaking session storage for a second reason - how are we differing between session storage (Document’s node navigable’s traversable navigable’s storage shed) and local storage (user agent level).

Reply:
Oh, no, sorry I understand now, we're checking if storageType is session inline:

```
    if (m_storage_type == StorageType::Local)
        return m_page->client().page_did_request_storage_keys("localStorage"_string, m_storage_key.to_string());
    return m_map.keys();
```

My confusion is in the context of https://github.com/LadybirdBrowser/ladybird/blob/6f01331b1f188dbb627cd6f4dc9bb86065b1c7e2/Libraries/LibWeb/StorageAPI/StorageBottle.cpp#L42 since `StorageJar` is effectively representing a storage shed, which makes https://github.com/LadybirdBrowser/ladybird/blob/3a4f2faf2e458b6670090c86a8645f5af0acf181/Libraries/LibWeb/StorageAPI/StorageShed.cpp#L33

Maybe there is a more natural way of doing it so that this difference remains in `obtain_a_storage_bottle_map`? 🤔

Reply:
sure, I split StorageBottle into LocalStorageBottle and SessionStorageBottle. `user_agent_storage_shed()` is also gone and for local storage we now construct bottle directly bypassing shed.

Reply:
Could we also be passing through the identifier from obtain_a_local_storage_bottle_map instead of repeating "localStorage"?

Reply:
sure, I went even further and replaced identifier type with enum. no reason to use a string for this value.